### PR TITLE
Update docs wrt. latest erdjs (add a short cookbook, as well) - Part 1

### PR DESCRIPTION
**Pull request by andreibancioiu**
#### Did you test the changes locally ?
- [x] yes
- [ ] no

#### Which category (categories) does this pull request belong to?
- [x] document new feature
- [x] update documentation that is not relevant anymore
- [x] add examples or more information about a component
- [ ] fix grammar issues
- [ ] other

### CODE REVIEW

File: docs/sdk-and-tools/erdjs/erdjs-cookbook.md [hunk 1]
```diff
@@ -0,0 +1,573 @@
+---
+id: erdjs-cookbook
+title: Cookbook
+---
+
+This page will guide you through the process of handling common tasks using **erdjs**.
+
+:::important
+This cookbook makes use of `erdjs 10`.
```
Comment (bogdan-rosianu) [inline docs/sdk-and-tools/erdjs/erdjs-cookbook.md:9]: please reference the migration guide here

File: docs/sdk-and-tools/erdjs/writing-and-running-erdjs-snippets.md [hunk 2]
```diff
@@ -3,10 +3,8 @@ id: writing-and-testing-erdjs-interactions
 title: Writing and testing interactions
 ---
 
-Writing and testing interactions
-
 :::important
-This tutorial makes use of `erdjs 10`, which isn't officially released yet (in _beta_ as of 2022-03-24). Therefore, the code examples in this section are subject to change.
+This tutorial makes use of `erdjs 10`.
```
Comment (bogdan-rosianu) [inline docs/sdk-and-tools/erdjs/writing-and-running-erdjs-snippets.md:7]: name the file as `writing-and-testing-erdjs-interactions.md` ?

**Review by bogdan-rosianu — CHANGES_REQUESTED** [review]

File: docs/sdk-and-tools/erdjs/erdjs.md [hunk 3]
```diff
@@ -5,9 +5,34 @@ title: erdjs
...
-It consists of TypeScript / JavaScript helpers and utilities for interacting with the Blockchain (in general) and with Smart Contracts (in particular).
+**erdjs** consists of TypeScript / JavaScript helpers and utilities for interacting with the Blockchain (in general) and with Smart Contracts (in particular).
+
+| Package | Source code | Description |
+| ------------- | ----------------------------------------- | ------------------------------------- |
+| [erdjs](https://www.npmjs.com/package/@elrondnetwork/erdjs) | [Github](https://github.com/ElrondNetwork/elrond-sdk-erdjs) | Basic components for interacting with the blockchain and with smart contracts. |
+| [erdjs-walletcore](https://www.npmjs.com/package/@elrondnetwork/erdjs-walletcore) | [Github](https://github.com/ElrondNetwork/elrond-sdk-erdjs-walletcore) | Core wallet components (generation, signing). |
+| [erdjs-contract-wrappers](https://www.npmjs.com/package/@elrondnetwork/erdjs-contract-wrappers) | [Github](https://github.com/ElrondNetwork/elrond-sdk-erdjs-contract-wrappers) | Smart Contract wrappers, based on erdjs. NodeJS repl-friendly. |
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply docs/sdk-and-tools/erdjs/erdjs.md:14]: Indeed. Removed link, until published.

File: docs/sdk-and-tools/erdjs/erdjs-cookbook.md [hunk 4]
```diff
@@ -0,0 +1,573 @@
...
+let aliceOnNetwork = await networkProvider.getAccount(addressOfAlice);
+alice.update(aliceOnNetwork);
+
+console.log(alice.nonce);
+console.log(alice.balance);
+```
+
+### Managing the sender nonce locally
+
+When sending a bunch of transactions, you usually have to first fetch the account nonce from the network (see above), then manage it locally (e.g. increment upon signing & broadcasting a transaction):
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply docs/sdk-and-tools/erdjs/erdjs-cookbook.md:62]: Referenced.

File: docs/sdk-and-tools/erdjs/erdjs-cookbook.md [hunk 5]
```diff
@@ -0,0 +1,573 @@
...
+let firstPayment = TokenPayment.egldFromAmount("1.5");
+let secondPayment = TokenPayment.egldFromBigInteger("1500000000000000000");
+console.log(firstPayment.valueOf(), secondPayment.valueOf());
+console.log(firstPayment.toPrettyString(), secondPayment.toPrettyString());
+```
+
+A `TokenPayment` object for transferring **fungible** tokens:
+
+```
+let firstPayment = TokenPayment.fungibleFromAmount(identifier, "1.5", numDecimals);
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply docs/sdk-and-tools/erdjs/erdjs-cookbook.md:92]: Good suggestion! Added.

File: docs/sdk-and-tools/erdjs/erdjs-cookbook.md [hunk 6]
```diff
@@ -0,0 +1,573 @@
...
+let quantity = 50;
+let payment = TokenPayment.semiFungible(identifier, nonce, quantity);
+```
+
+A `TokenPayment` object for transferring **non-fungible** tokens:
+
+```
+let nonce = 7;
+let payment = TokenPayment.nonFungible(identifier, nonce);
+```
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply docs/sdk-and-tools/erdjs/erdjs-cookbook.md:109]: Added.

File: docs/sdk-and-tools/erdjs/erdjs-cookbook.md [hunk 7]
```diff
@@ -0,0 +1,573 @@
...
+    data: new TransactionPayload("helloWorld"),
+    gasLimit: 70000,
+    receiver: new Address("erd1spyavw0956vq68xj8y4tenjpq2wd5a9p2c6j8gsz7ztyrnpxrruqzu66jx"),
+    value: TokenPayment.egldFromAmount(1),
+    chainID: "D"
+});
+```
+
+### Broadcast using a network provider
+
```
[the remark this answers is not in the document]
Comment (andreibancioiu) [reply docs/sdk-and-tools/erdjs/erdjs-cookbook.md:132]: Somehow fixed. No dedicated page on using the signing providers. Referenced dapp-core.

File: docs/sdk-and-tools/erdjs/erdjs-cookbook.md (hunk 1 above)
Comment (andreibancioiu) [reply docs/sdk-and-tools/erdjs/erdjs-cookbook.md:9]: Added.

File: docs/sdk-and-tools/erdjs/writing-and-running-erdjs-snippets.md (hunk 2 above)
Comment (andreibancioiu) [reply docs/sdk-and-tools/erdjs/writing-and-running-erdjs-snippets.md:7]: Actually, I wanted to coin the term `erdjs-snippets` (i.e. Mocha scenarios, runnable in VsCode with a single click): https://github.com/ElrondNetwork/erdjs-snippets.

Is it OK to keep the name?

**Pull request merged**